hicommonwealth/commonwealth
# Fix devnet tests

## Link to Issue
Closes: [#5707](https://github.com/hicommonwealth/commonwealth/issues/5707)

## Description of Changes
- Created the `execute-devnet-tests.sh` script which allows `yarn test-devnet` to take in a `cosmos` or `evm` argument
- Fixes a DB-related error with creating a test contract for CE 2.0 tests
- Fixes an interaction that CE 2.0 tests have with TBC via the Ganache chain node
- Removed an 'after' mocha function which was failing and is unnecessary
- Added `start-docker-containers.sh` which allows starting the Redis or RabbitMQ containers if Docker is set up with or without root access. This is because if you have docker setup with root access by default, using `sudo docker` commands will prompt for a password and then fail.
- Fixed cosmos test instability - courtesy of @mhagel

## "How We Fixed It"



## Test Plan
- Ensure that EVM and Cosmos devnet tests are executing AND passing

Run EVM tests locally
- Run `yarn --cwd packages/commonwealth/test/util/evm-chain-testing`
- Run `yarn --cwd packages/commonwealth/test/util/evm-chain-testing ganache`
- In another terminal run `yarn --cwd packages/commonwealth/test/util/evm-chain-testing start`
- In another terminal run `yarn --cwd packages/commonwealth test-devnet evm`

Run Cosmos tests locally
- Run `yarn --cwd packages/commonwealth reset-db && yarn --cwd packages/commonwealth migrate-db`
- Run `yarn --cwd packages/commonwealth cosmos:build`
- Run `yarn --cwd packages/commonwealth start`
- In another terminal run `yarn --cwd packages/commonwealth test-devnet cosmos`

## Deployment Plan

1. 

## Other Considerations

-
